{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[34m\u001b[1mtrain: \u001b[0mweights=yolov5s.pt, cfg=, data=data.yaml, hyp=../data/hyps/hyp.scratch-low.yaml, epochs=3, batch_size=16, imgsz=320, rect=False, resume=False, nosave=False, noval=False, noautoanchor=False, noplots=False, evolve=None, bucket=, cache=None, image_weights=False, device=, multi_scale=False, single_cls=False, optimizer=SGD, sync_bn=False, workers=8, project=../runs/train, name=exp, exist_ok=False, quad=False, cos_lr=False, label_smoothing=0.0, patience=100, freeze=[0], save_period=-1, local_rank=-1, entity=None, upload_dataset=False, bbox_interval=-1, artifact_alias=latest\n",
      "\u001b[34m\u001b[1mgithub: \u001b[0mup to date with git@github.com:mjaber95/FeedMe âœ…\n",
      "YOLOv5 ðŸš€ 0.4-16-g2832e85 Python-3.8.13 torch-1.11.0 CPU\n",
      "\n",
      "\u001b[34m\u001b[1mhyperparameters: \u001b[0mlr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.05, cls=0.5, cls_pw=1.0, obj=1.0, obj_pw=1.0, iou_t=0.2, anchor_t=4.0, fl_gamma=0.0, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0\n",
      "\u001b[34m\u001b[1mWeights & Biases: \u001b[0mrun 'pip install wandb' to automatically track and visualize YOLOv5 ðŸš€ runs (RECOMMENDED)\n",
      "\u001b[34m\u001b[1mTensorBoard: \u001b[0mStart with 'tensorboard --logdir ../runs/train', view at http://localhost:6006/\n",
      "Overriding model.yaml nc=80 with nc=30\n",
      "\n",
      "                 from  n    params  module                                  arguments                     \n",
      "  0                -1  1      3520  models.common.Conv                      [3, 32, 6, 2, 2]              \n",
      "  1                -1  1     18560  models.common.Conv                      [32, 64, 3, 2]                \n",
      "  2                -1  1     18816  models.common.C3                        [64, 64, 1]                   \n",
      "  3                -1  1     73984  models.common.Conv                      [64, 128, 3, 2]               \n",
      "  4                -1  2    115712  models.common.C3                        [128, 128, 2]                 \n",
      "  5                -1  1    295424  models.common.Conv                      [128, 256, 3, 2]              \n",
      "  6                -1  3    625152  models.common.C3                        [256, 256, 3]                 \n",
      "  7                -1  1   1180672  models.common.Conv                      [256, 512, 3, 2]              \n",
      "  8                -1  1   1182720  models.common.C3                        [512, 512, 1]                 \n",
      "  9                -1  1    656896  models.common.SPPF                      [512, 512, 5]                 \n",
      " 10                -1  1    131584  models.common.Conv                      [512, 256, 1, 1]              \n",
      " 11                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']          \n",
      " 12           [-1, 6]  1         0  models.common.Concat                    [1]                           \n",
      " 13                -1  1    361984  models.common.C3                        [512, 256, 1, False]          \n",
      " 14                -1  1     33024  models.common.Conv                      [256, 128, 1, 1]              \n",
      " 15                -1  1         0  torch.nn.modules.upsampling.Upsample    [None, 2, 'nearest']          \n",
      " 16           [-1, 4]  1         0  models.common.Concat                    [1]                           \n",
      " 17                -1  1     90880  models.common.C3                        [256, 128, 1, False]          \n",
      " 18                -1  1    147712  models.common.Conv                      [128, 128, 3, 2]              \n",
      " 19          [-1, 14]  1         0  models.common.Concat                    [1]                           \n",
      " 20                -1  1    296448  models.common.C3                        [256, 256, 1, False]          \n",
      " 21                -1  1    590336  models.common.Conv                      [256, 256, 3, 2]              \n",
      " 22          [-1, 10]  1         0  models.common.Concat                    [1]                           \n",
      " 23                -1  1   1182720  models.common.C3                        [512, 512, 1, False]          \n",
      " 24      [17, 20, 23]  1     94395  models.yolo.Detect                      [30, [[10, 13, 16, 30, 33, 23], [30, 61, 62, 45, 59, 119], [116, 90, 156, 198, 373, 326]], [128, 256, 512]]\n",
      "Model summary: 270 layers, 7100539 parameters, 7100539 gradients, 16.1 GFLOPs\n",
      "\n",
      "Transferred 343/349 items from yolov5s.pt\n",
      "Scaled weight_decay = 0.0005\n",
      "\u001b[34m\u001b[1moptimizer:\u001b[0m SGD with parameter groups 57 weight (no decay), 60 weight, 60 bias\n",
      "\u001b[34m\u001b[1mtrain: \u001b[0mScanning '/Users/mamdouhjaber/code/mjaber95/FeedMe/data/train/labels.cach\u001b[0m\n",
      "\u001b[34m\u001b[1mval: \u001b[0mScanning '/Users/mamdouhjaber/code/mjaber95/FeedMe/data/valid/labels.cache'\u001b[0m\n",
      "Plotting labels to ../runs/train/exp16/labels.jpg... \n",
      "\n",
      "\u001b[34m\u001b[1mAutoAnchor: \u001b[0m5.28 anchors/target, 1.000 Best Possible Recall (BPR). Current anchors are a good fit to dataset âœ…\n",
      "Image sizes 320 train, 320 val\n",
      "Using 8 dataloader workers\n",
      "Logging results to \u001b[1m../runs/train/exp16\u001b[0m\n",
      "Starting training for 3 epochs...\n",
      "\n",
      "     Epoch   gpu_mem       box       obj       cls    labels  img_size\n",
      "       0/2        0G     0.129   0.06568   0.08831       180       320: 100%|â–ˆâ–ˆâ–ˆ\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@\n",
      "                 all        103       1227    0.00108     0.0318   0.000636   0.000143\n",
      "\n",
      "     Epoch   gpu_mem       box       obj       cls    labels  img_size\n",
      "       1/2        0G    0.1113   0.07233   0.08616       247       320: 100%|â–ˆâ–ˆâ–ˆ\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 1.060s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 1.060s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 1.060s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 0.310s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@\n",
      "                 all        103         74      0.691     0.0115    0.00187   0.000376\n",
      "\n",
      "     Epoch   gpu_mem       box       obj       cls    labels  img_size\n",
      "       2/2        0G   0.09265   0.07044   0.08382       225       320: 100%|â–ˆâ–ˆâ–ˆ\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 1.060s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 1.060s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 1.060s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 0.310s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@\n",
      "                 all        103        122     0.0112      0.242      0.021    0.00542\n",
      "\n",
      "3 epochs completed in 0.092 hours.\n",
      "Optimizer stripped from ../runs/train/exp16/weights/last.pt, 14.4MB\n",
      "Optimizer stripped from ../runs/train/exp16/weights/best.pt, 14.4MB\n",
      "\n",
      "Validating ../runs/train/exp16/weights/best.pt...\n",
      "Fusing layers... \n",
      "Model summary: 213 layers, 7091035 parameters, 0 gradients, 16.1 GFLOPs\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 1.060s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 1.060s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 1.060s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@WARNING: NMS time limit 0.310s exceeded\n",
      "               Class     Images     Labels          P          R     mAP@.5 mAP@\n",
      "                 all        103        111     0.0444      0.235     0.0215     0.0057\n",
      "               apple        103          6          0          0     0.0213    0.00401\n",
      "              banana        103          6     0.0138      0.167      0.037    0.00445\n",
      "                beef        103          1          0          0          0          0\n",
      "         blueberries        103          4     0.0183          1      0.034    0.00847\n",
      "               bread        103          4          0          0          0          0\n",
      "              butter        103          2     0.0327        0.5     0.0659     0.0291\n",
      "              carrot        103          3          0          0          0          0\n",
      "              cheese        103          2          0          0          0          0\n",
      "             chicken        103          3    0.00507      0.333      0.085     0.0255\n",
      "      chicken_breast        103          1          0          0          0          0\n",
      "           chocolate        103          1          0          0          0          0\n",
      "                corn        103          3          0          0          0          0\n",
      "                eggs        103          3     0.0165      0.667     0.0282    0.00509\n",
      "               flour        103          3          0          0          0          0\n",
      "         goat_cheese        103          1          0          0          0          0\n",
      "         green_beans        103          5     0.0586        0.2     0.0433    0.00876\n",
      "         ground_beef        103          2     0.0115        0.5     0.0468    0.00937\n",
      "                 ham        103          1          0          0          0          0\n",
      "         heavy_cream        103          7          0          0          0          0\n",
      "                lime        103          4          0          0     0.0193    0.00578\n",
      "                milk        103          5          0          0          0          0\n",
      "           mushrooms        103          5     0.0213        0.6     0.0174    0.00331\n",
      "               onion        103          5          0          0          0          0\n",
      "              potato        103          6     0.0563      0.333      0.066     0.0153\n",
      "              shrimp        103          2     0.0202        0.5      0.015    0.00451\n",
      "             spinach        103          5          0          0          0          0\n",
      "        strawberries        103          6     0.0326          1      0.126     0.0352\n",
      "               sugar        103          5          1          0          0          0\n",
      "        sweet_potato        103          4     0.0298       0.75     0.0255    0.00742\n",
      "              tomato        103          6     0.0165        0.5     0.0132    0.00487\n",
      "Results saved to \u001b[1m../runs/train/exp16\u001b[0m\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n",
      "Exception ignored in: <function StorageWeakRef.__del__ at 0x105234280>\n",
      "Traceback (most recent call last):\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/multiprocessing/reductions.py\", line 36, in __del__\n",
      "  File \"/Users/mamdouhjaber/.pyenv/versions/3.8.13/envs/FeedMe/lib/python3.8/site-packages/torch/storage.py\", line 520, in _free_weak_ref\n",
      "AttributeError: 'NoneType' object has no attribute '_free_weak_ref'\n"
     ]
    }
   ],
   "source": [
    "!python ../train.py --img 320 --batch 16 --epochs 3 --data data.yaml --weights yolov5s.pt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "760a1187a787a144d8aefe46e7f6f24369e452e3b0916dbbe3b6587e70d2fb78"
  },
  "kernelspec": {
   "display_name": "Python 3.8.13 64-bit ('FeedMe')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.13"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
